### PR TITLE
PLATFORM-7930 update Subscription

### PR DESCRIPTION
**Pull request by mmeller-wikia**
https://fandom.atlassian.net/browse/PLATFORM-7930

### CODE REVIEW

File: src/Subscription.php [hunk 1]
```diff
@@ -92,104 +32,18 @@ public function hasSubscription( ?string $providerId = null ) {
...
-	 * @param bool|null $local [Optional] True or False to enable or disable.  Not passing this argument results in no change.
-	 *
-	 * @return bool Previous value, Enabled or Disabled
-	 */
-	public static function useLocalCacheOnly( $local = null ) {
-		$return = self::$useLocalCacheOnly; // Copy so the return value is the old value if being changed.
-		if ( is_bool( $local ) ) {
-			self::$useLocalCacheOnly = $local;
-		}
-		return $return;
```
Comment (mmeller-wikia) [inline src/Subscription.php]: Noop functions

File: src/SubscriptionHooks.php [hunk 2]
```diff
@@ -13,240 +13,68 @@
...
 
 	/**
-	 * Handle setting if the user requires HTTPS per subscription.
-	 *
-	 * @param object $user User
-	 * @param bool &$requiresHttps Requires HTTPS
-	 *
-	 * @return bool True
-	 */
-	public static function onUserRequiresHTTPS( $user, &$requiresHttps ) {
```
Comment (mmeller-wikia) [inline src/SubscriptionHooks.php]: Hooks has been removed. AFAIK we are enforcing HTTPS for all MW pages

File: src/SubscriptionHooks.php [hunk 3]
```diff
@@ -13,240 +13,68 @@
...
-					'httpOnly' => $config->get( 'CookieHttpOnly' )
-				]
-			);
-
-			$redirect = $request->getFullRequestURL();
-			if ( strpos( $request->getFullRequestURL(), 'https://' ) === 0 ) {
-				$redirect = substr_replace( $redirect, 'http://', 0, 8 );
-			}
-			$output->enableClientCache( false );
-			$output->redirect( $redirect, ( $request->wasPosted() ? '307' : '302' ) );
```
Comment (mmeller-wikia) [inline src/SubscriptionHooks.php]: This hook redirect users to HTTP. This functionality was disabled by flag `"FullHTTPSExperiment": true`

File: src/SubscriptionHooks.php [hunk 4]
```diff
@@ -13,240 +13,68 @@
...
-			// Do not mess with external redirects.
-			return true;
-		}
-
-		if ( $wgRequest->getProtocol() !== 'http' && strpos( $redirect, 'https://' ) === 0 ) {
-			$redirect = substr_replace( $redirect, 'http://', 0, 8 );
-			if ( $output->getRequest()->wasPosted() ) {
-				$code = '307';
-			}
-		}
```
Comment (mmeller-wikia) [inline src/SubscriptionHooks.php]: One more time messing with HTTP/HTTPS. Functionality disabled by `FullHTTPSExperiment` flag

File: src/SubscriptionHooks.php [hunk 5]
```diff
@@ -13,240 +13,68 @@
...
-			if ( $subscription !== false ) {
-				$_cacheSetting = Subscription::skipCache( true );
-				// Don't care about the return.  This just forces a recache.
-				$subscription->getSubscription();
-				Subscription::skipCache( $_cacheSetting );
-			}
-		}
-
-		return true;
-	}
```
Comment (mmeller-wikia) [inline src/SubscriptionHooks.php]: Noop function, cache wasn't implemented for subscription classes

File: src/SubscriptionProvider.php [hunk 6]
```diff
@@ -119,13 +85,4 @@ abstract public function cancelCompedSubscription( int $userId );
...
 	}
-
-	/**
-	 * Return the duration to cache API responses in seconds.
-	 *
-	 * @return int Duration to cache API responses in seconds.
-	 */
-	public function getCacheDuration() {
-		return 600; // Cache for ten minutes.
-	}
```
Comment (mmeller-wikia) [inline src/SubscriptionProvider.php]: Unused

**Review by maaartyyynaa — APPROVED** [review]

**Pull request merged**